test/hotspot/jtreg/compiler/types/TestMethodHandleSpeculation.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc.. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8269285
+ * @summary Crash/miscompile in CallGenerator::for_method_handle_inline after JDK-8191998
+ * @requires vm.compMode == "Xmixed" & vm.flavor == "server"
+ *
+ * @run main/othervm
+ *        -Xcomp -Xbatch
```

Comment:
The test can be sped up significatly by compiling `compiler.types.TestMethodHandleSpeculation::main()` only (`-XX:CompileCommand=quiet -XX:CompileCommand=compileonly,compiler.types.TestMethodHandleSpeculation::main`).

Reply:
I tried it before with `-XX:CompileOnly=compiler.types.TestMethodHandleSpeculation::*`, and IIRC that did not work. I suspect you need to compile/inline the lambda body too. After all, this is `CallGenerator::for_method_handle_inline`? I can try again.

Reply:
There's some difference in behavior between `-XX:CompileCommand=compileonly,` and `-XX:CompileOnly=`.
IIRC the former doesn't affect inlining while the latter does.

Reply:
Right, that works, hm. See update.